components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsFetchResourceImplTest.java

```diff
@@ -0,0 +1,294 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DefaultPatientsFetchResourceImplTest
+{
+    private static final String EID1 = "EID1";
+
+    private static final String EID2 = "EID2";
+
+    private static final String ID_1 = "P0000001";
+
+    private static final String ID_2 = "P0000002";
+
+    private static final String ID_3 = "P0000003";
+
+    private static final String ID_4 = "P0000004";
+
+    private static final String EID_LABEL = "eid";
+
+    private static final String ID_LABEL = "id";
+
+    private static final String XWIKI_LABEL = "xwiki";
+
+    private static final String DATA_LABEL = "data";
+
+    private static final String LINKS_LABEL = "links";
+
+    @Rule
+    public final MockitoComponentMockingRule<PatientsFetchResource> mocker =
+        new MockitoComponentMockingRule<PatientsFetchResource>(DefaultPatientsFetchResourceImpl.class);
+
+    @Mock
+    private Patient patient1;
+
+    @Mock
+    private Patient patient2;
+
+    @Mock
+    private Patient patient3;
+
+    @Mock
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private DefaultPatientsFetchResourceImpl component;
+
+    private Request request;
+
+    private Query query;
+
+    private final DocumentReference patientReference1 = new DocumentReference(XWIKI_LABEL, DATA_LABEL, ID_1);
+
+    private final DocumentReference patientReference2 = new DocumentReference(XWIKI_LABEL, DATA_LABEL, ID_2);
+
+    private final DocumentReference patientReference3 = new DocumentReference(XWIKI_LABEL, DATA_LABEL, ID_1);
+
+    private final JSONObject patient1JSON = new JSONObject().put(ID_LABEL, ID_1);
+
+    private final JSONObject patient2JSON = new JSONObject().put(ID_LABEL, ID_2);
+
+    private final JSONObject patient3JSON = new JSONObject().put(ID_LABEL, ID_3);
+
+    private final Collection<String> uriList = ImmutableList.of("http://uri");
+
+    @Before
+    public void setUp() throws ComponentLookupException, QueryException
+    {
+        MockitoAnnotations.initMocks(this);
+
+        final Execution execution = mock(Execution.class);
+        final ExecutionContext executionContext = mock(ExecutionContext.class);
+        final ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.component = (DefaultPatientsFetchResourceImpl) this.mocker.getComponentUnderTest();
+
+        this.logger = this.mocker.getMockedLogger();
+        this.repository = this.mocker.getInstance(PatientRepository.class, "secure");
+
+        when(this.patient1.getId()).thenReturn(ID_1);
+        when(this.patient1.getDocument()).thenReturn(this.patientReference1);
```

Comment:
Do we use these (all the patientReferenceXXX) anywhere? (I don't see any use but it may be used somewhere deeper which I don't see off the top of my head)

I'm asking because another PR (https://github.com/phenotips/phenotips/pull/2148) going into this release changes how we work with patient documents, so if we can remove this alltogether it will simplify merging the changes (or at least simplify figuring out if something has to be changed or not)